src/intermediate_config.rs

```diff
@@ -48,65 +53,76 @@ fn parse_config_from_str(config_content: &str) -> Result<IntermediateConfig, Str
         Ok(content) => content[0].to_owned()
     };
 
-    let remote_machine = match &yaml["remoteMachine"] {
+    let remote = match &yaml["remote"] {
         Yaml::Hash(remote_machine) => {
             let host = match &remote_machine.get(&Yaml::String(String::from("host"))) {
                 Some(host) => match host {
                     Yaml::String(host) => Some(host.to_string()),
                     Yaml::Null => None,
-                    _ => return Err(String::from("remoteMachine.host must be a string"))
+                    _ => return Err(String::from("remote.host must be a string"))
                 },
                 None => None
             };
 
-            Some(IntermediateRemoteMachine {
+            Some(IntermediateRemote {
                 host,
             })
         }
         Yaml::Null | Yaml::BadValue => None,
-        ref something_else => return Err(format!("'remoteMachine' must be an object, but was {:#?}", something_else))
+        ref something_else => return Err(format!("'remote' must be an object, but was {:#?}", something_else))
     };
 
-    let compression = match &yaml["compression"] {
-        Yaml::Hash(compression) => {
-            let local = match compression.get(&Yaml::String(String::from("local"))).cloned() {
-                Some(local) => match local {
-                    Yaml::Integer(local) => if local >= 1 && local <= 9 {
-                        Some(local)
+    let push = match &yaml["push"] {
```

Comment:
I think it is time to move `push` and `pull` reading to a single function since essentially they are the same. Maybe even replace `struct Pull` and `struct Push` with `struct Sync` or whatever.

Reply:
They're not same, `Pull` will support `parallel` mode soon, common fields can be extracted though